singularity-eos/eos/eos_base.hpp

```diff
@@ -34,6 +34,29 @@ static inline auto member_func_name(const char *type_name, const char *func_name
 namespace singularity {
 namespace eos_base {
 
+namespace impl {
+// Cuda doesn't have strcat, so we implement it ourselves
+PORTABLE_FORCEINLINE_FUNCTION
+char *StrCat(char *destination, const char *source) {
+  int i, j;
+
+  // move to the end of the destination string
+  // specifically avoid strlen, which isn't on GPU
+  for (i = 0; destination[i] != '\0'; i++)
+    ;
+  // Appends characters of the source to the destination string
+  for (j = 0; source[j] != '\0'; j++) {
+    destination[i + j] = source[j];
+  }
+
+  // null terminate destination string
+  destination[i + j] = '\0';
+
+  // the destination is returned by standard `strcat()`
+  return destination;
+}
```

Comment:
Thank you for doing this!

Since we're already rolling our own `StrCat()`, would it be reasonable to add explicit bound arguments for the char arrays and perform length checking or at least truncation?

Reply:
Yes I think this is a good idea. I'll add it.